Display error messages for creating and joining rooms

ntwbruce opened the pull request:
Currently the user is not informed of any errors encountered in situations where creating or joining a room fails.

Let's
- add a text box to the Lobby scene to display any potential errors
- add overriden OnCreateRoomFailed method to RoomCreator to update Lobby text box if a room with the user-given name already exists
- add overriden OnJoinRoomFailed method to RoomJoiner to update Lobby text box if a room with the user-given name does not exist, or if it is full

ntwbruce committed:
Display error messages for creating and joining rooms

Currently the user is not informed of any errors encountered in
situations where creating or joining a room fails.

Let's
- add a text box to the Lobby scene to display any potential errors
- add overriden OnCreateRoomFailed method to RoomCreator to update Lobby
text box if a room with the user-given name already exists
- add overriden OnJoinRoomFailed method to RoomJoiner to update Lobby
text box if a room with the user-given name does not exist, or if it is
full

lowkangn reviewed (approved):
LGTM

kenneth-fung commented on Assets/Scripts/Network/RoomJoiner.cs:
```diff
@@ -20,4 +22,13 @@ public override void OnJoinedRoom()
+
+    public override void OnJoinRoomFailed(short returnCode, string message) {
+        base.OnJoinRoomFailed(returnCode, message);
+        if (returnCode == 32758) {
```
Perhaps define this number as a `public static readonly int LOBBY_ERROR_NAME` field in either `RoomCreator` or `RoomJoiner` instead.

kenneth-fung commented on Assets/Scripts/Network/RoomJoiner.cs:
```diff
@@ -20,4 +22,13 @@ public override void OnJoinedRoom()
+        base.OnJoinRoomFailed(returnCode, message);
+        if (returnCode == 32758) {
+            errorMessage.GetComponent<Text>().text = "A lobby with that name does not exist!";
+        } else if (returnCode == 32765) {
```
Same here.

kenneth-fung reviewed (changes requested):
Aside from the comments, looks good. Go ahead with merge once changed.

kenneth-fung commented on Assets/Scripts/Network/RoomCreator.cs:
```diff
@@ -1,6 +1,7 @@
 using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine.UI;
```
Let's use the `TextMeshProUGUI` component instead of the Unity `Text` component for this message box, since the rest of the UI is using that already.

kenneth-fung commented on Assets/Scripts/Network/RoomJoiner.cs:
```diff
@@ -9,6 +10,7 @@ public class RoomJoiner : MonoBehaviourPunCallbacks
 {
     [SerializeField] private TMP_InputField roomNameInputField;
     [SerializeField] private string roomSceneName;
+    [SerializeField] private GameObject errorMessage;
```
Perhaps change the type here from `GameObject` to `TextMeshProUGUI` so that you don't have to use `GetComponent` on the GameObject when setting the text (which can be costly). You can still drag the same object into this field in the inspector.

lowkangn replied:
Teach me senpai

kenneth-fung replied:
No 🗿

kenneth-fung reviewed (changes requested):
My bad, some more minor changes.

ntwbruce committed:
Merge branch 'master' into branch-create-join-handler

# Conflicts:
#	Assets/Scenes/Lobby.unity
#	Assets/Scripts/Network/RoomCreator.cs
#	Assets/Scripts/Network/RoomJoiner.cs

ntwbruce committed:
Update input for Lobby's Display Error Message object Room Joiner script

ntwbruce committed:
Replace Display Error Message GameObject with TextMeshProUGUI component

ntwbruce committed:
Update RoomCreator and RoomJoiner

- Convert errorMessage GameObject to TextMeshProUGUI component
- Abstract out exit code values

ntwbruce committed:
Remove copy of Lobby scene

kenneth-fung reviewed (approved):
Looks good to me, thanks for the changes.
